src/ctap2/credential_management.rs

```diff
@@ -479,23 +484,8 @@ where
             .parent()
             // by construction, RK has a parent, its RP
             .unwrap();
+        self.delete_rp_dir_if_empty(rp_path);
```

Comment:
I'm not very familiar with fido-authenticator, but shouldn't this also be inserted in `delete_resident_key_by_user_id`?

Reply:
Currently, the `delete_resident_key_by_user_id` and `delete_resident_key_by_path` functions only perform the deletion itself.  Clean-up has to be down by the caller because these functions don’t know if it is necessary or not.  The alternative would be what I mentioned in the PR text:  Always performing the clean-up, even if this means that the directory is deleted and re-created directly afterwards.

Reply:
And since the cleanup is not atomic we need to support the case where it hasn't been done i see. This makes sense to me.